src/code/z_camera.c

```diff
@@ -450,58 +445,60 @@ f32 Camera_GetFloorYLayer(Camera* camera, Vec3f* norm, Vec3f* pos, s32* bgId) {
 }
 
 /**
- * Returns the CameraSettingType of the camera at index `camDataIdx`
+ * Returns the CameraSettingType of the camera at index `bgCamDataIndex`
  */
-s16 Camera_GetCamDataSetting(Camera* camera, s32 camDataIdx) {
-    return func_80041A4C(&camera->play->colCtx, camDataIdx, BGCHECK_SCENE);
+s16 Camera_GetSurfaceBgCamDataSetting(Camera* camera, s32 bgCamDataIndex) {
+    return BgCheck_GetBgCamDataSettingImpl(&camera->play->colCtx, bgCamDataIndex, BGCHECK_SCENE);
 }
 
 /**
  * Returns the scene camera info for the current camera data index
  */
-Vec3s* Camera_GetCamBGData(Camera* camera) {
-    return func_80041C10(&camera->play->colCtx, camera->camDataIdx, BGCHECK_SCENE);
+Vec3s* Camera_GetSurfaceBgCamData(Camera* camera) {
+    return BgCheck_GetBgCamDataVec3sImpl(&camera->play->colCtx, camera->bgCamDataIndex, BGCHECK_SCENE);
 }
 
 /**
  * Gets the scene's camera index for the poly `poly`, returns -1 if
  * there is no camera data for that poly.
  */
-s32 Camera_GetDataIdxForPoly(Camera* camera, s32* bgId, CollisionPoly* poly) {
-    s32 camDataIdx;
+s32 Camera_GetSurfaceBgCamDataIndex(Camera* camera, s32* bgId, CollisionPoly* poly) {
+    s32 bgCamDataIndex;
     PosRot playerPosRot;
     s32 ret;
 
     Actor_GetWorldPosShapeRot(&playerPosRot, &camera->player->actor); // unused.
-    camDataIdx = SurfaceType_GetCamDataIndex(&camera->play->colCtx, poly, *bgId);
+    bgCamDataIndex = SurfaceType_GetBgCamDataIndex(&camera->play->colCtx, poly, *bgId);
 
-    if (func_80041A4C(&camera->play->colCtx, camDataIdx, *bgId) == CAM_SET_NONE) {
+    if (BgCheck_GetBgCamDataSettingImpl(&camera->play->colCtx, bgCamDataIndex, *bgId) == CAM_SET_NONE) {
         ret = -1;
     } else {
-        ret = camDataIdx;
+        ret = bgCamDataIndex;
     }
     return ret;
 }
 
 /**
  * Returns the scene camera info for the floor under the player.
  * If there is no floor then there is no camera data, returns the number of
- * pieces of data there are in `dataCnt`
+ * pieces of data there are in `bgCamDataNumData`
  */
-Vec3s* Camera_GetCamBgDataUnderPlayer(Camera* camera, u16* dataCnt) {
+Vec3s* Camera_GetSurfaceBgCamDataAndNumData(Camera* camera, u16* bgCamDataNumData) {
```

Comment:
What about its sister function `Camera_GetWaterBoxDataIdx`/`Camera_GetWaterBoxBgCamDataIndex` ?

Reply:
something like `Camera_GetBgCamDataUnderPlayer` seems fine for surfaces.

`Camera_GetWaterBoxBgCamDataIndex` seems fine as is, but if we want to make it clear its about the waterbox that player is in, it could be `Camera_GetCurWaterBoxBgCamDataIndex`

since waterboxes are more scarce/in less abundance than a surface/poly, it feels less important to me to specify that its for a specific one. where as for a surface, the "under the player" part feels more important

Reply:
`Camera_GetBgCamDataUnderPlayer` is fine for me then. Interestingly, it's only used a single time and it's for crawlspaces.

I'd also be fine with `Camera_GetWaterBoxBgCamDataIndex` as is

Reply:
oh, yea that is weird that its only used once. How do other cases get the cam data from the floor poly under the player?